crates/sargon-uniffi/src/core/hash.rs

```diff
@@ -0,0 +1,54 @@
+use crate::prelude::*;
+use sargon::Exactly32Bytes as InternalExactly32Bytes;
+use sargon::Hash as InternalHash;
+
+/// Represents a 32-byte hash digest.
+///
+/// Made UniFFI convertible via HashSecretMagic,
+/// exposed in Swift/Kotlin as its own struct/data class, with
+/// hidden secret magic.
+#[derive(
+    Clone, PartialEq, Eq, std::hash::Hash, InternalConversion, uniffi::Record,
+)]
+pub struct Hash {
+    pub(crate) value: Exactly32Bytes,
+}
+
+impl From<InternalHash> for Hash {
```

Comment:
is the `impl From<InternalHash> for Hash`  and `impl Into<InternalHash> for Hash` needed ? Cannot `InternalConversion` proc macro generate that for us?

Reply:
maybe it can but you just have not done it yet?

Reply:
The macro cannot do it yet, to allow conversion of internal types that do not match in structure the types defined in uniffi. These are few, but definitely worth in future iterations to add such capability.